src/cookbooks/clients_last_seen_bits.md

```diff
@@ -0,0 +1,136 @@
+
+An powerful feature of the `clients_last_seen` methodology is that we can capture
+per-day usage information for the entire 28-day period that each row in the table
+describes. We encode usage information in "bit patterns" where the physical
+type of the field is a BigQuery 64-bit integer, but logically the integer
+represents an array of bits, with each 1 indicating a day where the given clients
+was active and each 0 indicating a day where the client was inactive.
+
+The representation is space-efficient and actually easier to manipulate
+in BigQuery than a date array would be, but requires some practice to interact
+with directly. For well-established usage and retention metrics, we generally
+provide logic in views to insulate users from having to ever consider the
+raw bit patterns, but new investigations may require novel retention definitions.
+
+## Usage: Backward-looking windows
+
+The simplest application of usage bit patterns is for calculating metrics in
+backward-looking windows. This is what we do for our canonical _usage_ measures
+DAU, WAU, and MAU.
+
+Let's imagine a single row from `clients_last_seen` with `submission_date = 2020-01-28`.
+The `days_seen_bits` field records usage for a single client over a period of 28 days
+_ending_ on the given `submission_date`. We will call this anchor date "day 0" 
+(2020-01-28 in this case) and count backwards to "day -27" (2020-01-01).
+
+Let's suppose this client was only active on two days in the past month:
+2020-01-22 (day -6) and 2020-01-15 (day -13). That client's`days_seen_bits`
+value would show up as an integer with value `8256` which is pretty inscrutable.
+Let's look at that value as a string representing each bit instead:
+
+```
+SELECT udf.bits28_to_bitstring(32768)
+-- 0000000000000010000001000000
+```
+
+Let's make that representation a bit more visual:
+
+```
+    0  0  0  0  0  0  0  0  0  0  0  0  0  0  1  0  0  0  0  0  0  1  0  0  0  0  0  0
+    ──────────────────────────────────────────────────────────────────────────────────
+    │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │  │
+    │ 26  │ 24  │ 22  │ 20  │ 18  │ 16  │ 14  │ 12  │ 10  │  8  │  6  │  4  │  2  │  0
+   27    25    23    21    19    17    15    13    11     9     7     5     3     1
+   
+  └──────────────────────────────────────────────────────────────────────────────────┘
+  MAU                                                             └──────────────────┘
+                                                                   WAU             └─┘
+                                                                                   DAU
+```
+
+In this picture, we've annotated the windows for DAU (day 0 only), 
+WAU (days 0 through -6) and MAU (days 0 through -27). This particular client
+won't count toward DAU for 2020-01-28, but the client does count towards both
+WAU and MAU.
+
+You may notice that the day 13 bit doesn't end up affecting these metrics.
```

Comment:
It does affect MAU, but the activity on day 6 also affects MAU so you don't 'need' to see them on the 13th to count them in this specific case.